typescript/sdk/src/deploy/verify/types.ts

```diff
@@ -7,22 +7,68 @@ export type ContractVerificationInput = {
 
 export type VerificationInput = ContractVerificationInput[];
 
+// see https://etherscan.io/contract-license-types
+export enum ExplorerLicenseType {
+  NO_LICENSE = '1',
+  UNLICENSED = '2',
+  MIT = '3',
+  GPL2 = '4',
+  GPL3 = '5',
+  LGPL2 = '6',
+  LGPL3 = '7',
+  BSD2 = '8',
+  BSD3 = '9',
+  MPL2 = '10',
+  OSL3 = '11',
+  APACHE2 = '12',
+  AGPL3 = '13',
+  BSL = '14',
+}
+
 export type CompilerOptions = {
   codeformat: 'solidity-standard-json-input';
-  compilerversion: string; // see https://etherscan.io/solcversions for list of support versions, inferred from build artifact
-  licenseType:
-    | '1'
-    | '2'
-    | '3'
-    | '4'
-    | '5'
-    | '6'
-    | '7'
-    | '8'
-    | '9'
-    | '10'
-    | '11'
-    | '12'
-    | '13'
-    | '14'; // integer from 1-14, see https://etherscan.io/contract-license-types
+  compilerversion: string; // see https://etherscan.io/solcversions for list of support versions
+  licenseType: ExplorerLicenseType;
 };
+
+export enum ExplorerApiActions {
+  GETSOURCECODE = 'getsourcecode',
+  VERIFY_IMPLEMENTATION = 'verifysourcecode',
+  MARK_PROXY = 'verifyproxycontract',
+  CHECK_STATUS = 'checkverifystatus',
+  CHECK_PROXY_STATUS = 'checkproxyverification',
+}
+
+export enum ExplorerApiErrors {
+  ALREADY_VERIFIED = 'Contract source code already verified',
+  ALREADY_VERIFIED_ALT = 'Already Verified',
+  VERIFICATION_PENDING = 'Pending in queue',
+  PROXY_FAILED = 'A corresponding implementation contract was unfortunately not detected for the proxy address.',
+  BYTECODE_MISMATCH = 'Fail - Unable to verify. Compiled contract deployment bytecode does NOT match the transaction deployment bytecode.',
+}
+
+export type FormOptions<Action extends ExplorerApiActions> =
+  Action extends ExplorerApiActions.GETSOURCECODE
+    ? {
```

Comment:
Nice use of conditional types here. A discriminated union may have been more legible but this is good too!